lib/matplotlib/tests/test_datetime.py

```diff
@@ -600,11 +600,28 @@ def test_violinplot(self):
         fig, ax = plt.subplots()
         ax.violinplot(...)
 
-    @pytest.mark.xfail(reason="Test for vlines not written yet")
     @mpl.style.context("default")
     def test_vlines(self):
-        fig, ax = plt.subplots()
-        ax.vlines(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        fig, (ax1, ax2, ax3) = plt.subplots(3, 1, layout='constrained')
+        dates = [datetime.datetime(2023, 2, 10),
+                 datetime.datetime(2023, 5, 18),
+                 datetime.datetime(2023, 6, 6)]
+        ax1.set_xlim(left=datetime.datetime(2023, 1, 1),
+                     right=datetime.datetime(2023, 6, 30))
+        ax2.set_xlim(left=0,
+                     right=0.5)
+        ax3.set_xlim(left=datetime.datetime(2023, 7, 1),
+                     right=datetime.datetime(2023, 12, 31))
+        ax1.vlines(x=dates,
+                   ymin=[0, 0.25, 0.5],
+                   ymax=[0.25, 0.5, 0.75])
+        ax2.vlines(x=[0.3, 0.4],
+                   ymin=[np.datetime64('2023-03-20'), np.datetime64('2023-03-31')],
+                   ymax=[np.datetime64('2023-05-01'), np.datetime64('2023-05-16')])
+        ax3.vlines(x=[datetime.datetime(2023, 9, 1), datetime.datetime(2023, 12, 10)],
+                   ymin=datetime.datetime(2023, 1, 15),
+                   ymax=datetime.datetime(2023, 1, 30))
```

Comment:
I would find it a bit clearer to group by Axes, not by method call.

Reply:
I was trying to follow the same format as test_axhline and test_axvline but I would agree that makes more sense. Just pushed the changes.

Reply:
Also do you know why my tests might be failing? When I run test_interactive_backends.py locally it seems to work with no issues. Any insight would be appreciated.

Reply:
It is unrelated #27412

Reply:
I see. Thank you